shared/middleware/net_sim_api.rb

```diff
@@ -361,6 +384,14 @@ def self.override_redis_for_test(override_redis)
 
   private
 
+  # Takes a block, providing a Redis client, and automates cleanup
+  def with_redis
+    client = get_redis_client
+    yield client
+  ensure
+    client.disconnect unless client.nil?
+  end
```

Comment:
Creating a helper that takes a block like this allows us to ensure that the client gets cleaned up when the block exits.

Note, `disconnect` is actually [a method on redis-slave-read](https://github.com/cheald/redis-slave-read/blob/master/lib/redis-slave-read/interface/hiredis.rb#L20) but not on the `Redis` instance (which instead has `disconnect!`).  I'm also not 100% sure this is the right method to call, there are a few options:

* [`Redis.quit`](https://github.com/redis/redis-rb/blob/29653bbc1e9d8ab6090107eacb2fde4d4a6af1aa/lib/redis.rb#L166) which does a round-trip to the server, and has a known behavior of opening a connection in order to send 'QUIT' when you had never connected in the first place.  Curiously, `redis-slave-read` [only forwards 'QUIT' to the master node](https://github.com/cheald/redis-slave-read/blob/master/lib/redis-slave-read/interface/hiredis.rb#L13).
* [`Redis.close`](https://github.com/redis/redis-rb/blob/29653bbc1e9d8ab6090107eacb2fde4d4a6af1aa/lib/redis.rb#L79) which does an immediate quiet disconnect, but is not supported by `redis-slave-read`
* [`Redis.disconnect!`](https://github.com/redis/redis-rb/blob/29653bbc1e9d8ab6090107eacb2fde4d4a6af1aa/lib/redis.rb#L82) which is just an alias to `Redis.close` and is also not supported by `redis-slave-read`.
* [`Hiredis.disconnect`](https://github.com/cheald/redis-slave-read/blob/master/lib/redis-slave-read/interface/hiredis.rb#L20) in `redis-slave-read`, which actually calls [`Redis.client.disconnect`](https://github.com/redis/redis-rb/blob/29653bbc1e9d8ab6090107eacb2fde4d4a6af1aa/lib/redis/client.rb#L240) on every node in the group.